src/Symfony/Component/HttpFoundation/Session/Storage/Handler/MongoDbSessionHandler.php

```
@@ -86,77 +93,97 @@ public function close(): bool
 
     protected function doDestroy(#[\SensitiveParameter] string $sessionId): bool
     {
-        $this->getCollection()->deleteOne([
-            $this->options['id_field'] => $sessionId,
-        ]);
+        $write = new BulkWrite();
+        $write->delete(
+            [$this->options['id_field'] => $sessionId],
+            ['limit' => 1]
+        );
+
+        $this->manager->executeBulkWrite($this->namespace, $write);
 
         return true;
     }
 
     public function gc(int $maxlifetime): int|false
     {
-        return $this->getCollection()->deleteMany([
-            $this->options['expiry_field'] => ['$lt' => new UTCDateTime()],
-        ])->getDeletedCount();
+        $write = new BulkWrite();
+        $write->delete(
+            [$this->options['expiry_field'] => ['$lt' => $this->getUTCDateTime()]],
+        );
+        $result = $this->manager->executeBulkWrite($this->namespace, $write);
+
+        return $result->getDeletedCount() ?? false;
```

Note: `null` is only [returned for unacknowledged writes](https://www.php.net/manual/en/mongodb-driver-writeresult.getdeletedcount.php), so the `false` case should not happen in practice here. Although I suppose it's possible if the user passed in such a Manager with `w=0` in its connection string.

And even beyond that, the static analysis tooling would probably require handling the `null` case anyway.

But beyond all of that, [SessionHandlerInterface](https://www.php.net/manual/en/sessionhandlerinterface.gc.php) docs note that `false` indicates _failure_. An unacknowledged write isn't a failure, so I reckon it'd only make sense to return `false` if you were maybe catching and suppressing an exception. Otherwise, defaulting to `0` seems sensible.

In the original code, `DeleteResult::getDeletedCount()` would have [raised a BadMethodCallException](https://github.com/mongodb/mongo-php-library/blob/1.16.1/src/DeleteResult.php#L49), which doesn't seem like a good user experience either.

---

Yes, that's what I've found too. But phpstan warns that `getDeletedCount` returns `int|null`, which doesn't match with the return type of the function `int|false`. So I did this to satisfy the static analysis.